### PR TITLE
Fix creation notification history records [MAILPOET-4679]

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

This PR fixes a possible memory issue with post notification that was unable to catch.

## Code review notes

_N/A_

## QA notes

Please, test that post notification sending works properly. Try scheduling immediately and also with the other frequency.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4679](https://mailpoet.atlassian.net/browse/MAILPOET-4679)

## After-merge notes

_N/A_

### CODE REVIEW

File: mailpoet/lib/Cron/Workers/Scheduler.php [hunk 1]
```diff
@@ -187,21 +187,31 @@ public function processPostNotificationNewsletter($newsletter, SendingTask $queu
...
       return $this->deleteQueueOrUpdateNextRunDate($queue, $newsletter);
     }
 
     // create a duplicate newsletter that acts as a history record
-    $notificationHistory = $this->createPostNotificationHistory($newsletterEntity);
+    try {
+      $notificationHistory = $this->createPostNotificationHistory($newsletterEntity);
+    } catch (\Exception $exception) {
+      $this->loggerFactory->getLogger(LoggerFactory::TOPIC_POST_NOTIFICATIONS)->error(
+        'creating post notification failed',
```
Comment (costasovo) [inline mailpoet/lib/Cron/Workers/Scheduler.php]: I think 'creating post notification failed' should be 'creating post notification history failed'.

**Review by costasovo — CHANGES_REQUESTED** [review]
Looks good. I have just one suggestion.

**Review by costasovo — APPROVED** [review]
LGTM! 👍

**Review by Aschepikov — APPROVED** [review]
Good job!

**Pull request merged**